pkg/controller/constraint/constraint_controller.go

```diff
@@ -111,6 +141,23 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	constraintKey := strings.Join([]string{instance.GetKind(), instance.GetName()}, "/")
+	enforcementAction, err := util.GetEnforcementAction(instance.Object)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	r.constraintsCache.addConstraintKey(constraintKey, tags{
+		enforcementAction: enforcementAction,
+		status:            activeStatus,
+	})
+	reportMetrics := true
+
+	defer func() {
+		if reportMetrics {
```

Comment:
This if statement is superfluous unless it's possible for reportMetrics to be false.

Reply:
added